dbms/src/Storages/Distributed/DistributedBlockOutputStream.cpp

```diff
@@ -80,6 +80,40 @@ void DistributedBlockOutputStream::writePrefix()
 
 
 void DistributedBlockOutputStream::write(const Block & block)
+{
+    auto has_materialized = [&]()
+    {
+        for (const auto & col : storage.getColumns().getMaterialized())
+            if (block.has(col.name))
+                return true;
+        return false;
+    };
+
+    if (has_materialized())
```

Comment:
A plain local bool variable would suffice for this. No need to write everything with lambdas -- it just leads to less clarity, slower compilation and bloated binary.

Nitpicks aside, do we even need this check? You have the same check in eraseMaterializedBlock, maybe we can just run it always?

Reply:
>it just leads to less clarity

Hm, for loop + var kinda looks less cleaner to me, but I don't have any objections, will fix this shortly

>do we even need this check?

This is to avoid overhead of the block copying

Reply:
Yep, the columns are copy-on-write ptrs, so we're not copying much.